src/VisualStudio/Core/Def/TableDataSource/DiagnosticTableItem.cs

```diff
@@ -44,10 +44,10 @@ public override TableItem WithAggregatedData(string[] projectNames, Guid[] proje
             => Data.ProjectId;
 
         public override LinePosition GetOriginalPosition()
-            => new(Data.DataLocation?.OriginalStartLine ?? 0, Data.DataLocation?.OriginalStartColumn ?? 0);
+            => Data.DataLocation?.OriginalFileSpan?.StartLinePosition ?? default;
 
         public override string? GetOriginalFilePath()
-            => Data.DataLocation?.OriginalFilePath;
+            => Data.DataLocation?.OriginalFileSpan?.Path;
```

Comment:
both the prior and new code are deeply unsettling to me.  It's unclear how these pieces of data flow all the way to the UI layer whenthey have non-sensical values.